c2pa_c_ffi/src/c_api.rs

```diff
@@ -431,6 +431,19 @@ pub unsafe extern "C" fn c2pa_string_free(s: *mut c_char) {
     }
 }
 
+/// Frees a const string allocated by Rust.
+///
+/// # Safety
+/// * Reads from NULL-terminated C strings.
+/// * The string can only be freed once and is invalid after this call.
+#[no_mangle]
+pub unsafe extern "C" fn c2pa_const_string_free(s: *const c_char) {
```

Comment:
We should just change c2pa_string_free to take a *const c_char. There is no valid case where it should be treated as a non constant. The bug here is that I didn't change c2pa_string_free to take a *const c_char when I updated the other functions to return one. We don't need a new function for this, we just need to fix the current one.